test/models/drive_test.dart

```diff
@@ -0,0 +1,1003 @@
+      expect(drive.endTime, DateTime.parse(json['end_time']));
+      expect(drive.seats, json['seats']);
+      expect(drive.cancelled, json['cancelled']);
+      expect(drive.driverId, json['driver_id']);
+      expect(drive.hideInListView, json['hide_in_list_view']);
+      expect(drive.driver == null, false);
+      expect(drive.rides!.length, 3);
+    });
+  });
+  group('Drive.fromJsonList', () {
+    test('parses a list of drives from json', () {
+      final Map<String, dynamic> json = {
+        'id': 43,
+        'created_at': '2021-01-01T00:00:00.000Z',
+        'start': 'London',
+        'start_lat': 2,
+        'start_lng': 3,
+        'start_time': '2022-01-01T00:00:00.000Z',
+        'end': 'Berlin',
+        'end_lat': 4,
+        'end_lng': 5,
+        'end_time': '2023-01-01T00:00:00.000Z',
+        'seats': 2,
+        'cancelled': true,
+        'driver_id': 7,
+        'hide_in_list_view': false,
+      };
+      final List<Map<String, dynamic>> jsonList = [json, json, json];
+      final List<Drive> drives = Drive.fromJsonList(jsonList);
+      expect(drives.length, 3);
+      expect(drives.first.id, json['id']);
+      expect(drives[1].start, json['start']);
+      expect(drives.last.seats, 2);
+    });
+    test('can handle an empty list', () {
+      final List<Drive> drives = Drive.fromJsonList([]);
+      expect(drives, []);
+    });
+  });
+  group('Drive.toJson', () {
+    test('returns a json representation of the drive', () async {
+      final Drive drive = DriveFactory().generateFake();
+      final Map<String, dynamic> json = drive.toJson();
+      expect(json['start'], drive.start);
+      expect(json['start_lat'], drive.startPosition.lat);
+      expect(json['start_lng'], drive.startPosition.lng);
+      expect(json['start_time'], drive.startTime.toString());
+      expect(json['end'], drive.end);
+      expect(json['end_lat'], drive.endPosition.lat);
+      expect(json['end_lng'], drive.endPosition.lng);
+      expect(json['end_time'], drive.endTime.toString());
+      expect(json['cancelled'], drive.cancelled);
+      expect(json['seats'], drive.seats);
+      expect(json['driver_id'], drive.driverId);
+      expect(json['hide_in_list_view'], drive.hideInListView);
+      expect(json.keys.length, 12);
+    });
+  });
+  group('Drive.approvedrides', () {
+    test('return null if rides is null', () async {
+      final Drive drive = DriveFactory().generateFake(
+        createDependencies: false,
+      );
+      expect(drive.approvedRides, null);
+    });
+    test('return an empty List if rides has no approved rides', () async {
+      final Drive drive = DriveFactory().generateFake(
+        rides: [
+          RideFactory().generateFake(status: RideStatus.pending),
+        ],
+      );
+      expect(drive.approvedRides, []);
+    });
+    test('return a ride in a List, if rides only has 1 ride and it is approved', () async {
```

Comment:
Do we need this test? One approved ride is already the test below this, and I don't have reason to expect the outcome of this to be different